app/controllers/api/v1/latest_photos_controller.rb

```diff
@@ -1,26 +1,21 @@
 class Api::V1::LatestPhotosController < ApplicationController
   def index
-    def index
-      @rover = Rover.find_by name: params[:rover_id].titleize
-      if @rover
-        render json: search_photos, each_serializer: PhotoSerializer, root: :latest_photos
-      else
-        render json: { errors: "Invalid Rover Name" }, status: :bad_request
-      end
-    end
-  end
+    rover = Rover.find_by name: params[:rover_id].titleize
 
-  private
+    if rover
+      validated_params = params
+        .permit(:rover_id, :camera, :earth_date, :size, :page, :per_page)
+        .merge(sol: rover.photos.maximum(:sol))
+      photos = helpers.search_photos rover, validated_params
 
-  def photo_params
-    params.permit(:camera, :earth_date, :rover_id).merge(sol: @rover.photos.maximum(:sol))
```

Comment:
I would leave these parameters in a `photo_params` method in order to keep the complexity of the `index` method down and to separate the responsibilities.